engine/api/workflow_queue.go

```diff
@@ -356,20 +357,24 @@ func (api *API) postSpawnInfosWorkflowJobHandler() service.AsynchronousHandler {
 		if err := UnmarshalBody(r, &s); err != nil {
 			return sdk.WrapError(err, "postSpawnInfosWorkflowJobHandler> cannot unmarshal request")
 		}
+		observability.Current(ctx, observability.Tag(observability.TagWorkflowNodeJobRun, id))
+		next()
 
 		tx, errBegin := api.mustDB().Begin()
 		if errBegin != nil {
 			return sdk.WrapError(errBegin, "postSpawnInfosWorkflowJobHandler> Cannot start transaction")
 		}
 		defer tx.Rollback()
 
+		_, next = observability.Span(ctx, "workflow.AddSpawnInfosNodeJobRun")
 		if err := workflow.AddSpawnInfosNodeJobRun(tx, id, s); err != nil {
 			return sdk.WrapError(err, "postSpawnInfosWorkflowJobHandler> Cannot save spawn info on node job run %d for %s name %s", id, getAgent(r), r.Header.Get(cdsclient.RequestedNameHeader))
```

Comment:
don't for to call the next() in the early returns

Reply:
done